test/test_ecmwf.py

```diff
@@ -0,0 +1,98 @@
+import os
+import mock
+import tempfile
+import shutil
+import json
+from mock import MagicMock
+
+from ecmwfapi import ECMWFDataServer
+
+from fetch.load import Config
+from fetch.ecmwf import EcmwfApiSource
+
+
+@mock.patch.dict(os.environ,{'HOME':tempfile.gettempdir()})
+@mock.patch('tempfile.mktemp', return_value='/path/to/fetch/dir-fetch')
+def test_ecmwf_retrieve_serialisation(mktemp_patch):
+    """
+    Tests the transformation from config to ecmwf retrieve calls
+    """
+    reporter = MagicMock()
+    raw_cfg = _make_ecmwf_config()
+    cfg = Config.from_dict(raw_cfg)
+
+    for item in cfg.rules:
+        with mock.patch('ecmwfapi.ECMWFDataServer') as MockServer:
+            mock_server = MockServer.return_value
+            src = item.module
+            src.trigger(reporter=reporter)
+
+            expected_args = {
+                'target': '/path/to/fetch/dir-fetch',
+                'class': getattr(raw_cfg['rules'][item.name]['source'], 'cls'),
+                'type': getattr(raw_cfg['rules'][item.name]['source'], 'typ')
+            }
+            expected_params = ['stream', 'area', 'levtype', 'expver', 'step',
+                               'dataset', 'grid', 'param', 'time', 'date']
+
+            for ep in expected_params:
+                if hasattr(raw_cfg['rules'][item.name]['source'], ep):
+                    expected_args[ep] = getattr(raw_cfg['rules'][item.name]['source'], ep)
+
+            mock_server.retrieve.assert_called_once_with(expected_args)
+
+
+def test_ecmwf_credentials_read():
+    """
+    Tests resolving credentials from the HOME folder
+    """
+
+    tmp_dir = tempfile.mkdtemp()
```

Comment:
Use a `tmp_path` (or `tmpdir` for older pytests without pathlib) argument in the test function to get a temp directory. 

pytest will handle the directory more intelligently (including cleanup): https://docs.pytest.org/en/latest/tmpdir.html